src/v/transform/logging/log_manager.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/property.h"
+#include "model/transform.h"
+#include "ssx/semaphore.h"
+#include "transform/logging/event.h"
+#include "transform/logging/io.h"
+#include "utils/absl_sstring_hash.h"
+#include "wasm/logger.h"
+
+#include <seastar/core/lowres_clock.hh>
+
+#include <absl/container/btree_map.h>
+
+#include <utility>
+
+namespace transform::logging {
+
+namespace detail {
+template<typename ClockType>
+class flusher;
+}
+
+template<typename ClockType = ss::lowres_clock>
+class manager {
```

Comment:
Can we add some documentation for the purpose and semantics of this class? Also details that it's expected to be an instance per core, etc.